situp-plugins/service-map-stateful/src/main/java/com/amazon/situp/plugins/processor/ServiceMapStatefulProcessor.java

```diff
@@ -39,28 +41,49 @@
     //TODO: Consider keeping this state in lmdb
     private volatile static  HashSet<ServiceMapRelationship> relationshipState = new HashSet<>();
     private static File databasePath;
-    //TODO: Strange errors when using the same db path. Could be because of different data types?
     private static File traceDatabasePath;
     private static Clock clock;
 
     private final int thisProcessorId;
 
-    public ServiceMapStatefulProcessor(final long windowDurationMillis, final File databasePath, final File traceDatabasePath, final int numProcessors,
+    public ServiceMapStatefulProcessor(final PluginSetting pluginSetting) {
+     this(pluginSetting.getIntegerOrDefault(ServiceMapProcessorConfig.WINDOW_DURATION, ServiceMapProcessorConfig.DEFAULT_WINDOW_DURATION)*TO_MILLIS,
+             new File(ServiceMapProcessorConfig.DEFAULT_SPANS_LMDB_PATH),
+             new File(ServiceMapProcessorConfig.DEFAULT_TRACES_LMDB_PATH),
+             Clock.systemUTC());
+    }
+
+    ServiceMapStatefulProcessor(final long windowDurationMillis,
+                                       final File databasePath,
+                                       final File traceDatabasePath,
                                        final Clock clock) {
-        this.clock = clock;
+        ServiceMapStatefulProcessor.clock = clock;
         this.thisProcessorId = processorsCreated.getAndIncrement();
         if(isMasterInstance()) {
```

Comment:
So if it is not the master instance the constructor does nothing? 
If thats the case it may be clearer to have an object hold all these variables and make that object an optional in this class. That would make it clearer that an instance of ServiceMapStatefulProcessor may or may not have a companion object with all these variables

Reply:
MasterInstance() is true when the first object of ServiceMapStatefulProcessor is created. This is actually not perfect, but it will work for now. Ideally we would want to split the shared state to a separate class, but that would be a bigger change. So will do it later.

No Optional in the entire project :)